src/sparseml/pytorch/image_classification/export.py

```diff
@@ -368,7 +386,13 @@ def main(
 
     if recipe is not None:
         ScheduledModifierManager.from_yaml(recipe).apply_structure(model)
+
+    if checkpoint_path:
         load_model(checkpoint_path, model, strict=True)
+
+    if one_shot is not None:
```

Comment:
Yes, both can exist, recipe is generally a training recipe, at some point in our codebase we moved to saving the training recipe in the checkpoint itself like `{'state_dict': ..., 'recipe': ...}` the training recipe being provided separately is to support older flows;

This one shot export flow, is meant so that someone can quickly test how the model will look like after application of another recipe.

for example, the original training recipe could quantize some parts of the graph; later if a user wants to apply another recipe on top that quantizes different layers or weights, they could pass in the `new` recipe with the one shot argument; if they saved the checkpoint without training recipe they could pass in the `old` training recipe with the `--recipe` argument;
like the following:
```
--recipe old_training_recipe.md --one-shot new_one_shot_recipe.md
```

What would happen here essentially is that the training recipe will be used to apply structure to the model then load weights, after which the one shot recipe will be applied and the model will be exported. Does this make sense?